src/api/app/jobs/update_released_binaries_job.rb

```diff
@@ -36,109 +36,107 @@ def perform(event_id)
   #   - sets BinaryRelease.obsolete_time
   #
   def update_binary_releases_for_repository(repository, new_binary_releases, time = Time.now)
-    # building a hash to avoid single SQL select calls slowing us down too much
+    # We record BinaryRelease attributes in this hash of hashes to be able to compare them later on.
+    # This is an optimization so we do not have to fetch BinaryRelease individually from the database.
+    # FIXME: This is exactly what ActiveRecord::Associations::CollectionProxy does no?
     old_binary_releases = {}
+
+    # We record all the BinaryRelease we process in this method to mark all but the current_binary_releases
+    # as "obsolete" at the end.
+    current_binary_releases = {}
+
+    # The Backend expresses the association between BinaryRelease via medium/ismedium:
+    #   - A BinaryRelease on a medium has the attribute `medium: 'openSUSE_Leap'`.
+    #   - A BinaryRelease that is a medium has the attribute `ismedium: 'openSUSE_Leap'`.
+    # We use this hash to build the BinaryRelease.on_medium relationship from the backend data
+    medium_hash = {}
+
+    # FIXME: This job is the only thing that ever changes the BinaryRelease table. It runs in it's own queue
+    #        with one worker. Only one UpdateReleasedBinariesJob will run at once. Why do we need this transaction?
     BinaryRelease.transaction do
+      # Populate the old_binary_releases hash
       repository.binary_releases.current.unchanged.find_each do |binary|
```

Comment:
the limit override is missing again, this will increase easily at least one hour of runtime again.

Reply:
What is a "limit override"?